src/main/java/pix/parser/Parser.java

```diff
@@ -0,0 +1,260 @@
+            return new ListDate(date);
+        } catch (DateTimeParseException e) {
+            throw new PixException("Please enter a date.\n\n e.g. listdate 'yyyy-MM-dd'");
+        }
+    }
+
+    /**
+     * Parses mark command.
+     *
+     * @param commandArr Index to be marked.
+     * @return Mark object.
+     * @throws PixException If index not given or index not in valid range.
+     */
+    public Command parseMark(String[] commandArr) throws PixException {
+        if (commandArr.length <= 1) {
+            throw new PixException("Please enter an index to mark.");
+        }
+        try {
+            int id = Integer.parseInt(commandArr[1]);
+            if (id > data.len() || id < 0) {
+                throw new PixException("Please enter a valid number.");
+            }
+            return new Mark(id - 1);
+        } catch (NumberFormatException e) {
+            throw new PixException(("Please enter an index to mark."));
+        }
+    }
+
+    /**
+     * Parses unmark command.
+     *
+     * @param commandArr Index to be un-marked.
+     * @return Unmark object.
+     * @throws PixException If index not given or index not in valid range.
+     */
+    public Command parseUnmark(String[] commandArr) throws PixException {
+        if (commandArr.length <= 1) {
+            throw new PixException("Please enter a index to unmark.");
+        }
+        try {
+            int id = Integer.parseInt(commandArr[1]);
+            if (id > data.len() || id < 0) {
+                throw new PixException("Please enter a valid number.");
+            }
+            return new Unmark(id - 1);
+        } catch (NumberFormatException e) {
+            throw new PixException(("Please enter an index to mark."));
+        }
+    }
+
+    /**
+     * Parses to do command.
+     *
+     * @param command Command entered by the user.
+     * @return Add To Do object.
+     * @throws PixException If no description given.
+     */
+    public Command parseTodo(String command) throws PixException {
+        String description = removeCommand(command);
+        if (description.isEmpty()) {
+            throw new PixException("Cannot have an empty task.");
+        }
+        return new AddToDo(description);
+    }
+
+    /**
+     * Parses deadline command.
+     *
+     * @param slashed Array of description and due date.
+     * @return AddDeadline object.
+     * @throws PixException If no description, due date or invalid format of date given.
+     */
+    public Command parseDeadline(String[] slashed) throws PixException {
+        if (slashed.length != 2 || removeCommand(slashed[0]).isEmpty() || removeCommand(slashed[1]).isEmpty()) {
+            throw new PixException("Invalid format.\n\n"
+                    + "e.g. deadline 'description' / 'yyyy-MM-dd HH-mm'");
+        }
+        try {
+            return new AddDeadline(removeCommand(slashed[0]), removeCommand(slashed[1]));

```

Comment:
Same as above comment, since removeCommand(Slashed[0]) and removeCommand(Slashed[1]) is referred to twice, would it be neater if we initialize these 2 objects with a more readable name and calling to it when needed?